cryptoconditions/types/rsa.py

```diff
@@ -0,0 +1,240 @@
+import math
+from base64 import urlsafe_b64decode
+
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPublicNumbers
+
+#from pyasn1.codec.der.decoder import decode as der_decode
+from pyasn1.codec.der.encoder import encode as der_encode
+from pyasn1.codec.native.decoder import decode as nat_decode
+#from pyasn1.codec.native.encoder import encode as nat_encode
+#
+#from cryptoconditions.condition import Condition
+#from cryptoconditions.fulfillment import Fulfillment
+from cryptoconditions.crypto import base64_add_padding
+from cryptoconditions.exceptions import MissingDataError, ValidationError
+from cryptoconditions.types.base_sha256 import BaseSha256
+from cryptoconditions.schemas.fingerprint import RsaFingerprintContents
+
+# Instantiate RSA signer with standard settings
+#rsa = Rsa()
+PUBLIC_EXPONENT = 65537
+#SALT_LENGTH = padding.PSS.MAX_LENGTH
+SALT_LENGTH = 32
+
+class RsaSha256(BaseSha256):
+    """RSA-SHA-256: RSA signature condition using SHA-256.
+
+    This RSA condition uses RSA-PSS padding with SHA-256. The salt length is set
+    equal the digest length of 32 bytes.
+
+    The public exponent is fixed at 65537 and the public modulus must be between
+    128 (1017 bits) and 512 bytes (4096 bits) long.
+
+    RSA-SHA-256 is assigned the type ID 3. It relies on the SHA-256 and RSA-PSS
+    feature suites which corresponds to a feature bitmask of 0x11.
+
+    """
+    TYPE_ID = 3
+    TYPE_NAME = 'rsa-sha-256'
+    TYPE_ASN1 = 'rsaSha256'
+    TYPE_ASN1_CONDITION = 'rsaSha256Condition'
+    TYPE_ASN1_FULFILLMENT = 'rsaSha256Fulfillment'
+    TYPE_CATEGORY = 'simple'
+
+    COST_RIGHT_SHIFT = 6    # 2**6 = 64
+
+    def __init__(self, asn1=None):
```

Comment:
I wonder why this implementation differs so much from the `ed25519` subtype where I could pass the public_key/modulus and signature in the `__init__` method.